app/models/service_template.rb

```diff
@@ -290,4 +322,33 @@ def validate_order
   def provision_action
     resource_actions.find_by(:action => "Provision")
   end
+
+  def set_resource_actions(ae_endpoints)
+    ae_endpoints ||= {}
+    [
+      {:name      => 'Provision',
+       :param_key => :provisioning,
+       :method    => 'default_provisioning_entry_point',
+       :args      => [service_type]},
+      {:name      => 'Reconfigure',
+       :param_key => :reconfigure,
+       :method    => 'default_reconfiguration_entry_point',
+       :args      => []},
+      {:name      => 'Retirement',
+       :param_key => :retirement,
+       :method    => 'default_retirement_entry_point',
+       :args      => []}
+    ].each do |action|
+      ae_endpoint = ae_endpoints[action[:param_key]] || {}
```

Comment:
skip if `ae_endpoint` is `nil`
If user wants to define an action without dialog and use default entry point, an empty hash should be used like
```
:config_info => {
  :retirement => {}
}
```